telegram/user.py

```diff
@@ -23,33 +23,29 @@
 
 
 class User(TelegramObject):
-    """This object represents a Telegram User.
+    """
+    This object represents a Telegram user or bot.
 
     Attributes:
-        id (int): Unique identifier for this user or bot
-        first_name (str): User's or bot's first name
-        last_name (str): User's or bot's last name
-        username (str): User's or bot's username
-        language_code (str): IETF language tag of the user's language
-        type (str): Deprecated
+        id (:obj:`int`): Unique identifier for this user or bot.
+        first_name (:obj:`str`): User's or bot's first name.
+        last_name (:obj:`str`): Optional. User's or bot's last name.
+        username (:obj:`str`): Optional. User's or bot's last name.
+        language_code (:obj:`str`): Optional. IETF language tag of the user's language.
+        bot (:class:`telegram.Bot`): Optional. The Bot to use for instance methods.
 
     Args:
-        id (int): Unique identifier for this user or bot
-        first_name (str): User's or bot's first name
-        **kwargs: Arbitrary keyword arguments.
-
-    Keyword Args:
-        type (Optional[str]): Deprecated
-        last_name (Optional[str]): User's or bot's last name
-        username (Optional[str]): User's or bot's username
-        language_code (Optional[str]): IETF language tag of the user's language
-        bot (Optional[telegram.Bot]): The Bot to use for instance methods
+        id (:obj:`int`): Unique identifier for this user or bot.
+        first_name (:obj:`str`): User's or bot's first name.
+        last_name (:obj:`str`, optional): User's or bot's last name.
+        username (:obj:`str`, optional): User's or bot's username.
+        language_code (:obj:`str`, optional): IETF language tag of the user's language.
+        bot (:class:`telegram.Bot`, optional): The Bot to use for instance methods.
     """
 
     def __init__(self,
                  id,
                  first_name,
-                 type=None,
```

Comment:
It does not break tests and users have no type, so I see no problem in just removing it. It's still captured with **kwargs if needed.